src/ui/App.tsx

```diff
@@ -27,6 +27,105 @@ enum LoadState {
   loaded,
 }
 
+// TODO: Add setaside
+enum PriorityVisibility {
+  visible,
+  hidden,
+}
+
+enum BuiltInAction {
+  archive,
+  moveToPriority,
+}
+
+// If the action value is a string, it's the name of the priority to move to.
+type Action = BuiltInAction | string;
+
+enum TriggerType {
+  deadline,
+  newMessage,
+}
+
+interface DeadlineTrigger {
+  type: TriggerType.deadline;
+  action: Action;
+  allowCustomDeadlines: boolean;
+  deadline?: number;
+}
+
+interface NewMessageTrigger {
+  type: TriggerType.newMessage;
+  action: Action;
+}
+
+interface PriorityConfig {
+  name: string;
+  visibility: PriorityVisibility;
+  triggers?: (DeadlineTrigger | NewMessageTrigger)[];
+}
+
+// OJAN: Prefix all the priority names with tm/ or mk/ as appropriate.
+const priorityConfigList: PriorityConfig[] = [
+  {
+    name: 'Mute',
+    visibility: PriorityVisibility.hidden,
+    triggers: [{ type: TriggerType.newMessage, action: BuiltInAction.archive }],
+  },
+  {
+    name: 'SoftMute/%epoch%',
+    visibility: PriorityVisibility.hidden,
+    triggers: [
+      // OJAN: This needs to maintain the epoch number
+      { type: TriggerType.newMessage, action: 'SoftMuteNewMessages/%epoch%' },
+      { type: TriggerType.deadline, action: BuiltInAction.archive, allowCustomDeadlines: true },
+    ],
+  },
+  {
+    name: 'Stuck',
+    visibility: PriorityVisibility.visible,
+  },
+  {
+    name: 'Stuck/%epoch%',
+    visibility: PriorityVisibility.hidden,
+    triggers: [
+      { type: TriggerType.newMessage, action: 'Stuck' },
+      { type: TriggerType.deadline, action: 'Stuck', allowCustomDeadlines: true },
```

Comment:
Is the "action" here correct?

Reply:
Yeah, see above that Stuck without the trailing epoch is PriorityVisibility.visible. So, this is making it visible in the inbox again.

Although, now that I write that, I realize that I'm merging two concepts into visible. There are visible-and-triaged and needs-triage. Those need to be separate visibility values.